### PR TITLE
bpo-36307: Upgrade Travis CI to Xenial and rm obsolete sudo:false

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Currently, the .travis.yml config file still has a ``sudo: false`` key, which previously triggered the Container-based environment on Travis. However, that was [removed in December 2018](https://docs.travis-ci.com/user/reference/overview/#deprecated-virtualization-environments), and so the key is now obsolete. Furthermore, the builds are still running in the Trusty (Ubuntu 14.04) environment, which will be at the end of its [5-year development lifecycle in April 2019 ](https://blog.ubuntu.com/2019/02/05/ubuntu-14-04-trusty-tahr) (next month). Therefore, ``sudo: false`` should be removed, and the presumably, if possible, the distribution should be upgraded to the modern ``dist: xenial`` (Ubuntu 16.04).

https://bugs.python.org/issue36307

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
skip news

**Comment by CAM-Gerlach** [comment]
Okay, looks like everything's green except for the allowed failures, which fail in the same way as on other PRs with the previous infrastructure. As a minor, 2-line meta-change to the CI infrastructure, I didn't think this merited a NEWS entry, but I'm happy to add one if desired.

**Review by Mariatta — APPROVED** [review]

**Comment by tirkarthi** [comment]
There were Openssl compatibility issues in the past and xenial used to be less stable. I would request waiting for people working closely with buildbots to review this before merging and also backporting since this changes base image of one of primary CI. See also some discussion in https://bugs.python.org/issue34009 .

**Comment by CAM-Gerlach** [comment]
Thanks @tirkarthi . 

> There were Openssl compatibility issues in the past

To note, at least according to the issue you linked, these issues were reported with *Trusty*, whereas the OpenSSL version distributed with Xenial is compatible. Therefore, we could switch back to the distro-provided version, though it makes sense to keep it building its own version to demonstrate how to do so, exercise that functionality and ensure compatibility with the latest OpenSSL.

While were at it, should we also bump the version up to the latest 1.1.1b, which is the current LTS branch (with 1.0.2 EoL in Dec. 2019 and 1.1.0 EoL in a few months, in September)?

>  xenial used to be less stable

We'll indeed want to confirm with the buildbot team and any other relevant parties, as this is not a trivial change, but I do note just based on the thread, the one individual who commented about Xenial not working reliably later took it back, saying it only required extra configuration (which is no longer the case). However, there may be other issues of which I'm aware (we ourselves haven't had any since spyder-ide/spyder#8802 , but obviously we're not exactly CPython).

**Comment by tirkarthi** [comment]
Thanks for the details. I was curious and just used search where the thread came up. I too would be happy to have it clarified along with backporting by someone else for the upgrade.

**Comment by CAM-Gerlach** [comment]
I searched for related terms as well, but for whatever reason didn't find that particular thread. Thanks for the heads up.

**Pull request merged**
